### ISSUE TITLE
Summary file naming could lead to files being overwritten

### ISSUE THREAD

**Issue by desilinguist** [EducationalTestingService/skll#37]
Let's say I run an experiment which generates a summary file. I then decide to run another experiment which uses a subset of the features my previous experiment used but with the same learners, the same training and test data, and the same settings. The summary file generated by the second experiment will overwrite the summary file from the first experiment since the following components are the same:
- training set
- test set
- scaling 
- tuning objective
- task
